ocaml/libs/xapi-stdext/lib/xapi-stdext-unix/unixext.ml

```diff
@@ -586,6 +586,55 @@ let with_polly_wait kind fd f =
       in
       f wait fd
 
+module Timeout = struct
+  type t = Mtime.Span.t
+
+  let of_span s : t = s
+
+  let of_string str =
+    let seconds = Float.of_string str in
+    match Mtime.Span.of_float_ns (seconds *. 1e9 |> Float.round) with
+    | Some span ->
+        span
+    | None ->
+        (* e.g. negative, or >=2^53. *)
+        invalid_arg (Printf.sprintf "Span out of range: %gs" seconds)
+
+  (* nanosecond conversion doesn't roundtrip, so use only microseconds, which does *)
+  let to_string t = Mtime.Span.to_float_ns t *. 1e-9 |> Printf.sprintf "%.6f"
+
+  let pp = Mtime.Span.pp
+end
+
+module Timer = struct
+  let delay dt = Mtime.Span.to_float_ns dt *. 1e-9 |> Thread.delay
+
+  type t = {elapsed: Mtime_clock.counter; timeout: Mtime.Span.t}
+
+  type remaining = Spare of Mtime.Span.t | Excess of Mtime.Span.t
+
+  let start ~timeout = {elapsed= Mtime_clock.counter (); timeout}
+
+  let elapsed {elapsed; _} = elapsed
```

Comment:
Users shouldn't be given access to the counter, IMO:
```suggestion
  let elapsed t = Mtime_clock.count t.elapsed
```

Reply:
The function is not exported, its only user is `pp`. I can move it inside `pp` to make that clearer.